tests/unit/defrag.cpp

```diff
@@ -67,18 +76,23 @@ int iterate_callback(struct eblob_disk_control *dc,
 	// Maybe it will be more comfortable with BOOST_TEST and error messages in more modern boost version.
 	BOOST_REQUIRE(!(dc->flags & BLOB_DISK_CTL_REMOVE));  // removed dc occured
 
-	iterator_private &ipriv = *static_cast<iterator_private*>(priv);
+	iterator_private &ipriv = *static_cast<iterator_private *>(priv);
+	if (ipriv.expect_blob_sorted) {
+		BOOST_REQUIRE_GT(data_offset, ipriv.prev_offset);
+		ipriv.prev_offset = data_offset;
+	}
+
 	auto &items = ipriv.items;
-	BOOST_REQUIRE(ipriv.number_checked < items.size()); // index out of range
+	BOOST_REQUIRE_LT(ipriv.number_checked, items.size()); // index out of range
```

Comment:
It is 4th refactoring, about replacing `BOOST_REQUIRE` on more specific checkers.